cmd/okctl/delete.go

```diff
@@ -170,8 +172,10 @@ including VPC, this is a highly destructive operation.`,
 			}
 
 			if delzones == "true" {
-				err = services.Domain.DeletePrimaryHostedZone(o.Ctx, o.CloudProvider, client.DeletePrimaryHostedZoneOpts{
-					ID: id,
+				var hz *state.HostedZone
+				err = services.Domain.DeletePrimaryHostedZone(o.Ctx, client.DeletePrimaryHostedZoneOpts{
+					ID:           id,
+					HostedZoneID: hz.ID,
```

Comment:
Won't this trigger a nullpointer exception? Since hz (on line 175) is not initialized

Reply:
Ugh .... yes - I have no idea how this could have worked, need to test and rewrite some more.